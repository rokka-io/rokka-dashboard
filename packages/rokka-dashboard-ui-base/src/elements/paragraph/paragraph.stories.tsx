import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import styled from 'styled-components';
import { colors } from '../../identity/colors/colors';

import { ParagraphLoginBrand } from './paragraph';

const ExampleContainer = styled.div`
  position: relative;
  background: ${colors.brand.primary};
  width: 200px;
  height: 200px;
`;

storiesOf('Elements / Paragraphs', module).add('Login Brand', () => (
  <ExampleContainer>
    <ParagraphLoginBrand>{text('text', 'Powered by Liip.')}</ParagraphLoginBrand>
  </ExampleContainer>
));
