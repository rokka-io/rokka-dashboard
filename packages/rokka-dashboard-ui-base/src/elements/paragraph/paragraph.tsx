import styled from 'styled-components';
import { colors } from '../../identity/colors/colors';
import { fonts } from '../../identity/typography/index';

export const ParagraphLoginBrand = styled.p`
  font-size: ${fonts.Sizes.small};
  color: ${colors.tints.white};
  position: absolute;
  bottom: 32px;
  right: 44px;
`;

export const ParagraphWhite = styled.p`
  color: ${colors.tints.white};
`;

export const SpanWhite = styled.span`
  color: ${colors.tints.white};
`;
