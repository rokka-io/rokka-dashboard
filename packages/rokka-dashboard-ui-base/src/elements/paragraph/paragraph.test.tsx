import { shallow } from 'enzyme';
import React from 'react';

import { ParagraphLoginBrand } from './paragraph';

describe('Elements', () => {
  describe('Paragraphs', () => {
    it('renders a login brand paragraph correctly.', () => {
      const component = shallow(<ParagraphLoginBrand>test</ParagraphLoginBrand>);

      expect(component).toMatchSnapshot();
    });
  });
});
