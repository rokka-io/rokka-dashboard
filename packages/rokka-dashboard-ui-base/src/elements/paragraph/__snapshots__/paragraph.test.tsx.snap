// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements Paragraphs renders a login brand paragraph correctly. 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  font-size: ",
          "1rem",
          ";
  color: ",
          "#fff",
          ";
  position: absolute;
  bottom: 32px;
  right: 44px;
",
        ],
      },
      "displayName": "styled.p",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "p",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  test
</StyledComponent>
`;
